src/components/Pagination/hooks/usePagination.ts

```diff
@@ -0,0 +1,59 @@
+import React, { useMemo } from "react";
+import { BaseUsePaginationProps } from "../type";
+
+export const DOTS = "...";
+
+// WA: Create Array with certain range based on data length
+export const range = (start: number, end: number) => {
+  let length = end - start + 1;
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+export const usePagination = (props: BaseUsePaginationProps) => {
```

Comment:
very sick-looking self-made hooks 😄!

tapi ini pake BaseUsePaginationProps, gapapa tuh hib?

Reply:
i think better suited name should be: UsePaginationType or UsePaginationProps, props kan refer ke component, i think better naming should be type. but, it's your pick!

Reply:
sure, iya naming nya gue rasa juga kurang pas. Okee soon diganti ya. Thanks!

Reply:
okeh!